File: CStoreManagement/app/src/main/java/com/c_store/zhiyazhang/cstoremanagement/view/interfaceview/ContractView.java
package com.c_store.zhiyazhang.cstoremanagement.view.interfaceview;

import android.view.MotionEvent;
import android.view.View;

import com.c_store.zhiyazhang.cstoremanagement.bean.ContractBean;
import com.c_store.zhiyazhang.cstoremanagement.bean.ContractTypeBean;
import com.c_store.zhiyazhang.cstoremanagement.bean.UserBean;
import com.c_store.zhiyazhang.cstoremanagement.presenter.contract.ContractAdapter;

import java.util.List;

/**
 * Created by zhiya.zhang
 * on 2017/5/10 14:34.
 */

public interface ContractView {

    //得到是否是通过搜索查询
    boolean isSearch();

    //如果是通过搜索查询到的就在此获得搜索关键字
    String getSearchMessage();

    //得到查的是哪个类
    ContractTypeBean getContractType();

    //监听短按一下
    void toShortClick(View view, ContractBean cb);

    //显示加载中
    void showLoading();

    //隐藏加载中
    void hideLoading();

    //把数据展示出来
    void showView(ContractAdapter adapter);

    //显示错误
    void showFailedError(String errorMessage);

    //长按加
    void touchAdd(ContractBean cb, int positon, MotionEvent event);

    //长按减
    void touchLess(ContractBean cb, int positon, MotionEvent event);

    //排序
    String getSort();

    //得到当前页数
    int getPage();

    //写入当前页数
    void setPage(int page);

    //上拉加载
    void pullLoading(ContractAdapter adapter);

    //把没信息提示显示出来
    void showNoMessage();

    //得到要修改的数据
    List<ContractBean> getContractList();

    //更新成功
    void updateDone();

}

File: KotlinCStoreManagement/app/src/main/java/com/cstore/zhiyazhang/cstoremanagement/view/interfaceview/ContractView.kt
package com.cstore.zhiyazhang.cstoremanagement.view.interfaceview

import android.view.MotionEvent
import com.cstore.zhiyazhang.cstoremanagement.bean.ContractBean
import com.cstore.zhiyazhang.cstoremanagement.bean.ContractTypeBean
import com.cstore.zhiyazhang.cstoremanagement.presenter.contract.ContractAdapter

/**
 * Created by zhiya.zhang
 * on 2017/6/14 15:52.
 */
interface ContractView {
    /**
     * 得到是否是通过搜索查询的
     */
    val isSearch: Boolean
    /**
     * 得到是否是得到所有改过的商品
     */
    val isJustLook: Boolean
    /**
     * 得到搜索的关键字
     */
    val searchMsg: String
    /**
     * 得到查看的类
     */
    val contractType: ContractTypeBean
    /**
     * 得到是什么排序
     */
    val sort: String

    /**
     * 得到、写入当前页
     */
    fun getPage(): Int

    fun setPage(value: Int)

    /**
     * 订量修改加
     */
    fun touchAdd(cb: ContractBean, event: MotionEvent, position: Int)

    /**
     * 订量修改减
     */
    fun touchLess(cb: ContractBean, event: MotionEvent, position: Int)

    /**
     * 上拉加载
     */
    fun pullLoading(adapter: ContractAdapter)

    /**
     * 显示无信息
     */
    fun showNoMessage()

    /**
     * 得到要修改的数据
     */
    val contractList: List<ContractBean>

    fun removeChangeDate()

    /**
     * 更新成功
     */
    fun updateDone()

    fun clickImage(cb: ContractBean, position: Int)
}
